src/client/dfuse/pil4dfs/int_dfs.c

```diff
@@ -1017,6 +1168,28 @@ consume_low_fd(void)
 			DS_ERROR(errno, "failed to reserve a low fd");
 			goto err;
 		} else if (low_fd_list[low_fd_count] >= DAOS_MIN_FD) {
+			/* reserve fd 255 too if unused. 255 is commonly used by bash. */
```

Comment:
This new code should go outside the while loop as it only needs to be executed once.

What this PR is doing currently is defining a fd at 1001, opening it if it's not already open, then using the fd and closing it at the end if it was able to open it.  What isn't handled is if the fd is not available, L1179 will close the new fd (which would likely be 1002) but the code will then use 1001 still which is A. a fd used by something else and B. possible not still open at the time it's used.

I think the best here would be to have a define of `#MIN_INTERNAL_FD` and a global of `dummy_fd` rather than a define, then when you open a file you can call `fnctl(..., F_DUPFD, MIN_INTERNAL_FD);` to get a custom FD and use this where needed, then close it at the end without the extra `fd_dummy_reserved` variable.